spec/unit/provider/package/windows/exe_package_spec.rb

```diff
@@ -71,10 +71,27 @@
 
   context '#install_command' do
     it 'should install using the source' do
+      allow(Puppet::FileSystem).to receive(:exist?).with(source).and_return(true)
       cmd = described_class.install_command({:source => source})
 
       expect(cmd).to eq(source)
     end
+
+    it 'should raise error when URI is invalid' do
+      web_source = 'https://www.t e s t.test/test.exe'
+
+      expect do
+        described_class.install_command({:source => web_source, :name => name})
+      end.to raise_error(Puppet::Error, /Error when installing #{name}, parsing URI:/)
+    end
+
+    it 'should download package from source file before installing' do
+      web_source = 'https://www.test.test/test.exe'
+      stub_request(:get, web_source).to_return(status: 200, body: "package binaries")
+      cmd = described_class.install_command({:source => web_source})
+      cmd = cmd.tr('\\', '/') # windows package returns windows path, switch to linux path
+      expect(File.read(cmd)).to eq('package binaries')
```

Comment:
`File.read` should be able to handle forward slashes on Windows. Is `munge(file_location)` adding an extra level of escaping?